tests/lib/rules/no-array-prototype-extensions.js

```diff
@@ -848,22 +848,31 @@ something.sort((a, b) => {
       code: 'something.sortBy(getKey())',
       output: `import { get } from '@ember/object';
 import { compare } from '@ember/utils';
-something.sort((a, b) => {
-            const key = getKey();
+[...something].sort((a, b) => compare(get(a, getKey()), get(b, getKey())))`,
```

Comment:
We should still keep `const key = getKey();` because we don't want to call `getKey()` multiple times. That could theoretically be an expensive operation or yield different results each call, so it's safer to only call it once.

Reply:
@bmish In most of the cases that i encountered, the passed param would be either a string or variable. In case if it is a function and is an expensive operation, i believe the developer can take care of assigning it to a variable and consume the variable. Avoiding a variable leads to a simpler code. wdyt?

Reply:
Gotcha. Made the necessary changes. Can you please take a look?